src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -373,6 +373,52 @@ void LegacyInferRequest::SetBatch(int new_batch) {
     }
 }
 
+void LegacyInferRequest::changeDefaultPtr() {
+    // renew external pointers before infer
+    const auto &inMap = graph->inputNodesMap;
+    for (auto it : inMap) {
+        auto name = it.first;
+        auto pBlob = MemoryDescUtils::interpretAsBlob(graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory());
+        InferenceEngine::TensorDesc desc = pBlob->getTensorDesc();
+        tuneInputDesc(name, desc);
+        if (externalPtr.count(name) &&
+            pBlob->getTensorDesc() == desc &&
+            graph->_normalizePreprocMap.find(name) == graph->_normalizePreprocMap.end() &&
+            !graph->getProperty().batchLimit) {
```

Comment:
The conditions here seems to be redundant duplication of what have been done in `::GetBlob()`, and it's unlikely to get a different result by re-checking them again here.  also suppose the checks succeeded in `::GetBlob()` but failed here, what should we do to `externalPtr[name]`? should we delete it from `externalPtr` to prevent `changeDefaultPtr()` from using it as input memory?

Anyway I don't think the re-evaluation of these conditions here would come up with different result. so why not simplify the condition here as following? 

```
if (externalPtr.count(name) && externalPtr[name] != _inputs[name]->buffer())
```

Reply:
aleady updated pr